cmd/net-rpc-client.go

```diff
@@ -137,7 +137,7 @@ func (rpcClient *RPCClient) Call(serviceMethod string, args interface{}, reply i
 	// rpc.Client for a subsequent reconnect.
 	err := rpcLocalStack.Call(serviceMethod, args, reply)
 	if err != nil {
-		if err.Error() == rpc.ErrShutdown.Error() {
+		if err == rpc.ErrShutdown {
```

Comment:
GIven that type is not lost if err is of type fmt.Errorf() and rpc.ErrShutdown is errors.New() they are not comparable even if the function returns with error interface compatible. This is the whole reason we have toStorageErr()

In this example you can see if Call() gets wrapped by something else you would loose that information.  

https://play.golang.org/p/6W8Sj7_5c_ 

Another interesting bug if you rely on this is shown here https://gist.github.com/harshavardhana/c53bb4b2efcdf3d66a2c